Assets/Scripts/SS3D/Systems/Health/BodyParts/Bodypart.cs

```diff
@@ -202,7 +191,7 @@ public void SetParentBodyPart(BodyPart value)
     /// Implementation should handle instantiating _bodyPartItem, removing the bodypart game object and doing whatever else is necessary.
     /// </summary>
     [Server]
-    private void DetachBodyPart()
+    protected void DetachBodyPart()
```

Comment:
I'd like to keep this method private, as it's gonna get a complete mess very soon if we don't. can you maybe instead use the events, or the abstract methods to do what you want to do ?